### ISSUE TITLE
Connection authentication

### ISSUE THREAD

**Issue by randomizedthinking** [JuliaDB/PostgreSQL.jl#41]
The current implementation of the `connect` function demands the password to make a connection. Is it a better approach to read .pgpass and respect the peer authentication setting?

**Comment by iamed2** [comment]
Hmm I didn't know about this. 

Well the correct way to handle it would be to change the password argument to a Nullable and pass C_NULL when no password is provided. `libpq` will then read `.pgpass` for the password.

**Comment by randomizedthinking** [comment]
It seems like a better way. Is it easy to do? I am new to julia while it sounds like a good starting point.

**Comment by iamed2** [comment]
Yup, just involves changing a few method signatures/calls. Give it a shot! :)
